meta2v2/meta2_utils.c

```diff
@@ -832,14 +842,26 @@ m2db_set_properties(struct sqlx_sqlite3_s *sq3, struct oio_url_s *url,
 
 	const char *name = ALIASES_get_alias(alias)->str;
 	gint64 version = ALIASES_get_version(alias);
+	// Used to remove duplicate modified properties
+	GHashTable *modified = g_hash_table_new_full(
+			g_str_hash, g_str_equal, NULL, _bean_clean);
 
 	if (flush) {
-		const char *sql = "alias = ? AND version = ?";
-		GVariant *params[3] = {NULL, NULL, NULL};
-		params[0] = g_variant_new_string (name);
-		params[1] = g_variant_new_int64 (ALIASES_get_version(alias));
-		err = _db_delete (&descr_struct_PROPERTIES, sq3->db, sql, params);
-		metautils_gvariant_unrefv (params);
+		GSList *deleted = NULL;
+		gchar *namev = NULL;
+		err = m2db_del_properties(sq3, url, &namev, &deleted);
+		for (GSList *l = deleted; !err && l; l = l->next) {
+			if (DESCR(l->data) == &descr_struct_PROPERTIES) {
+				struct bean_PROPERTIES_s *prop = l->data;
+				g_hash_table_replace(
+						modified, PROPERTIES_get_key(prop)->str, prop);
```

Comment:
We should replace the value of the property a null value.
```
$ openio object set toto magic --property test1=1 --property test2=2
$ openio object set toto magic --property test1=1 --clear
```
```
avril 01 11:14:29 openio-adu OIO,OPENIO,event-agent,1[15775]: 15775 7F2C104615F0 log ERROR {'job_id': '23', 'url': {'account': 'myaccount', 'content': '25B5EE3C7485050048CE7758603BFC07', 'version': '1554109168530733', 'user': 'toto', 'path': 'magic', 'ns': 'OPENIO', 'id': '8B78B3245B74710F3ACC1BEF4978E621F5E764E01FFB5621D23C4EECA2B7BB3D'}, 'when': 1554110069105128, 'parts': 1, 'request_id': '415852EB61F66304AE77CD59480189AC', 'part': 0, 'data': [{'alias': 'magic', 'version': 1554109168530733, 'type': 'properties', 'value': '1', 'key': 'test1'}, {'alias': 'magic', 'version': 1554109168530733, 'type': 'properties', 'value': '2', 'key': 'test2'}, {'name': 'magic', 'deleted': False, 'header': '25B5EE3C7485050048CE7758603BFC07', 'version': 1554109168530733, 'mtime': 1554109168, 'type': 'aliases', 'ctime': 1554109168}], 'event': 'storage.content.update'}
avril 01 11:14:38 openio-adu OIO,OPENIO,event-agent,1[15775]: 15775 7F2C104539B0 log ERROR {'job_id': '24', 'url': {'account': 'myaccount', 'content': '25B5EE3C7485050048CE7758603BFC07', 'version': '1554109168530733', 'user': 'toto', 'path': 'magic', 'ns': 'OPENIO', 'id': '8B78B3245B74710F3ACC1BEF4978E621F5E764E01FFB5621D23C4EECA2B7BB3D'}, 'when': 1554110078535160, 'parts': 1, 'request_id': '4171E2E76960BBBD9A5A5534DCE11E71', 'part': 0, 'data': [{'alias': 'magic', 'version': 1554109168530733, 'type': 'properties', 'value': '1', 'key': 'test1'}, {'alias': 'magic', 'version': 1554109168530733, 'type': 'properties', 'value': '2', 'key': 'test2'}, {'name': 'magic', 'deleted': False, 'header': '25B5EE3C7485050048CE7758603BFC07', 'version': 1554109168530733, 'mtime': 1554109168, 'type': 'aliases', 'ctime': 1554109168}], 'event': 'storage.content.update'}
```

Reply:
I have modified `m2db_del_properties` to null out properties before filling the list of deleted ones.